packages/wallet/src/redux/states/channels/funding.ts

```diff
@@ -1,235 +1,169 @@
-  return { type: B_WAIT_FOR_POST_FUND_SETUP, stage: FUNDING, ...channelOpen(params) };
+export function bWaitForPostFundSetup<T extends BaseFundingChannelState>(
+  params: T,
+): BWaitForPostFundSetup {
+  return { type: B_WAIT_FOR_POST_FUND_SETUP, stage: FUNDING, ...fundingChannelState(params) };
 }
 
-export function acknowledgeFundingSuccess<T extends ChannelOpen>(
+export function acknowledgeFundingSuccess<T extends BaseFundingChannelState>(
   params: T,
 ): AcknowledgeFundingSuccess {
-  return { type: ACKNOWLEDGE_FUNDING_SUCCESS, stage: FUNDING, ...channelOpen(params) };
+  return { type: ACKNOWLEDGE_FUNDING_SUCCESS, stage: FUNDING, ...fundingChannelState(params) };
 }
 
-export function sendFundingDeclinedMessage<T extends ChannelOpen>(
+export function sendFundingDeclinedMessage<T extends BaseFundingChannelState>(
   params: T,
 ): SendFundingDeclinedMessage {
-  return { type: SEND_FUNDING_DECLINED_MESSAGE, stage: FUNDING, ...channelOpen(params) };
+  return { type: SEND_FUNDING_DECLINED_MESSAGE, stage: FUNDING, ...fundingChannelState(params) };
 }
 
-export function acknowledgeFundingDeclined<T extends ChannelOpen>(
+export function acknowledgeFundingDeclined<T extends BaseFundingChannelState>(
   params: T,
 ): AcknowledgeFundingDeclined {
-  return { type: ACKNOWLEDGE_FUNDING_DECLINED, stage: FUNDING, ...channelOpen(params) };
+  return { type: ACKNOWLEDGE_FUNDING_DECLINED, stage: FUNDING, ...fundingChannelState(params) };
 }
 
-export function aDepositTransactionFailed<T extends ChannelOpen>(
-  params: T,
-): ADepositTransactionFailed {
-  return { type: A_DEPOSIT_TRANSACTION_FAILED, stage: FUNDING, ...channelOpen(params) };
-}
-export function bDepositTransactionFailed<T extends ChannelOpen>(
-  params: T,
-): BDepositTransactionFailed {
-  return { type: B_DEPOSIT_TRANSACTION_FAILED, stage: FUNDING, ...channelOpen(params) };
-}
-
-export type FundingState =
+export type FundingChannelState =
   | WaitForFundingRequest
-  | ApproveFunding
-  | AWaitForDepositToBeSentToMetaMask
-  | ASubmitDepositInMetaMask
-  | BWaitForOpponentDeposit
-  | AWaitForDepositConfirmation
-  | BWaitForDepositToBeSentToMetaMask
-  | BSubmitDepositInMetaMask
-  | AWaitForOpponentDeposit
-  | BWaitForDepositConfirmation
-  | BWaitForPostFundSetup
+  | WaitForFundingApproval
+  | WaitForFundingAndPostFundSetup
+  | WaitForFundingConfirmation
+  // ^^^ PlayerA should never let PlayerB get into this state, as it lets PlayerB move to the application phase with no real value at stake, giving them a chance to take the real value that's backing PlayerA's stake.
   | AWaitForPostFundSetup
-  | AcknowledgeFundingSuccess
+  | BWaitForPostFundSetup
+  | AcknowledgeFundingSuccess // <-- at this point, a message might need to be sent
   | SendFundingDeclinedMessage
-  | AcknowledgeFundingDeclined
-  | ADepositTransactionFailed
-  | BDepositTransactionFailed;
+  | AcknowledgeFundingDeclined;
+
+/**
```

Comment:
@andrewgordstewart Do we want to keep this comment here?

Reply:
Probably not!